BRANCHING.md

```diff
@@ -0,0 +1,66 @@
+
+### Directions for merging your branch `foo_COMP`
+
+- There is a CI job called `merges cleanly to develop` which runs whenever you have a PR off of `compatible`.  If that CI job passes, then you can simply merge `foo_COMP` into `compatible` , and no further action is needed.
+- If `merges cleanly to develop` does NOT pass, then when you’re done with your changes to `foo_COMP` and the PR is all approved, make a new branch+PR based off of `foo_COMP` called `foo_DEVELOP` (for example), and then merge `develop` into `foo_DEVELOP`.  Fix any merge errors that result, then once `foo_DEVELOP` is approved, you can merge it into `develop`.
+- Once `foo_DEVELOP` is fully landed into `develop`, then go and manually re-run the `merges cleanly to develop` CI job in the original `foo_COMP` PR.  The CI job should now pass and go green.  You can now merge `foo_COMP`
+- If after making `foo_DEVELOP`, you need to make changes to `foo_COMP`, then make sure to merge the newly updated `foo_COMP` into `foo_DEVELOP`.  In order for the git magic to work, `foo_DEVELOP` needs to be a superset of the commits from `foo`, and it also needs to merge first.  You can make further changes post-merge in `foo_DEVELOP` as needed to ensure correctness
+
+### Releases
+
+(**TODO** is this outdated/needs to be removed?)
+
+`release/1.X.X` branches are made off of `compatible` and tagged with alpha and beta tags until the code is deemed stable, then the `release/1.X.X` branch is merged into `master` and given a stable tag. Whenever code is tagged, if anything is missing in in the upstream branches (compatible, develop) then the tagged branch is also merged back for consistency.
+
+`release/2.0.0`is the branch where Berkeley QA Net releases are being cut, between `compatible`and `develop` in the tree. So far nothing has been tagged there but there will be `2.0.0alphaX` tags once the code is more stable and we are closer to the Incentivized testnet.
+
+Unless it is an emergency, code should flow from feature branches into `compatible` then in batches into the release branch for tagging and testing
+
+# proof-systems Branching policy
+Generally, proof-systems intends to be synchronized with the mina repository, and so its branching policy is quite similar. However several important distinctions exist that are a result of proof-systems having been created after the mainnet release (**TODO** ?).
+
+- `compatible`: **TODO** ??? Apparently not used in `mina`, and does not have commits after February 2023. Abandoned?
+    - Mina's `compatible`, similarly to mina's `master`, does not have `proof-systems`.
+- `develop`: matches mina's `develop`, soft fork-compatibility.
+  - small nuance: `proof-systems#develop` is used as a main branch for o1js.
+      - **TODO** clarify, this seems to not match with https://github.com/o1-labs/o1js/blob/main/README-dev.md#branch-compatibility?
```

Comment:
the branch that currently matches `o1js/main` in proof-systems is `proof-systems/o1js-main`. it would probably make sense to make that mapping official. using `proof-systems/develop` + `mina/o1js-main` won't work reliably